_drafts/practical_zephyr_05_west.md

```diff
@@ -0,0 +1,1246 @@
+FATAL ERROR: required program openocd not found; install it or add its location to PATH
+```
+
+Oh, the STM32 development kit uses a different _runner_ than the nRF52840 development kit: Nordic's development kits use `jlink`, whereas this development kit uses `openocd`.
+
+Thus, you don't only need the host tools and Zephyr SDK for your target, but also all other supporting tools, such as Nordic's command line tools or in this case `openocd`. On my computer, a simple `brew install open-ocd` fixes this problem, and I'm able to flash the application to the device.
+
+In a more serious setup, your entire toolchain should be maintained in a much better way, but - as promised - I won't try to tell how you should do that - it's entirely up to you.
+
+
+
+## Zephyr as dependency
+
+When browsing allowed imports, we've seen that Zephyr has dependencies. What we haven't seen yet, is that Zephyr itself can be a dependency: Depending on your application, you might make use of some frameworks or SDKs that in turn depend on Zephyr and thus list it as _project_ in their own `west.yml` file.
+
+E.g., looking at the [GitHub repository the _nRF Connect SDK_](https://github.com/nrfconnect/sdk-nrf/), we find that Nordic uses their own fork of the official Zephyr repository and has it listed as a _project_ in the SDK's [`west.yml`](https://github.com/nrfconnect/sdk-nrf/blob/e464540b74daca7f4a660f04fee0886b04365f70/west.yml#L62) file:
+
+[`https://github.com/nrfconnect/sdk-nrf/blob/main/west.yml`](https://github.com/nrfconnect/sdk-nrf/blob/main/west.yml)
+
+```yaml
+manifest:
+  projects:
+    - name: zephyr
+      repo-path: sdk-zephyr
+      revision: 90a72daae2c8715d760d974a7d294aa2eb6b38c4
+      import:
+        name-allowlist:
+          - TraceRecorderSource
+          - canopennode
+          - chre
+          - ...
+```
+
+Another example is the [Golioth Zephyr SDK](https://docs.golioth.io/firmware/zephyr-device-sdk/), which lists the official Zephyr repository as a project in their [`west-zephyr.yml`](https://github.com/golioth/golioth-zephyr-sdk/blob/4f0fcea1370bfec84f6754caaca8b3f35bbc3573/west-zephyr.yml#L3) manifest:
+
+[`https://github.com/golioth/golioth-zephyr-sdk/blob/main/west-zephyr.yml`](https://github.com/golioth/golioth-zephyr-sdk/blob/main/west-zephyr.yml)
+
+```yaml
+manifest:
+  projects:
+    - name: zephyr
+      revision: v3.5.0
+      url: https://github.com/zephyrproject-rtos/zephyr
+      west-commands: scripts/west-commands.yml
+      import: true
+```
+
+These dependencies are something that you need to handle yourself since [_West_ ignores projects that have already been defined in other files](https://docs.zephyrproject.org/latest/develop/west/manifest.html#manifest-imports). E.g., if you define _Zephyr_ as a project in your manifest file, and _after_ it add [Golioth's Zephyr SDK](https://docs.golioth.io/firmware/zephyr-device-sdk/), then the _Zephyr_ project in the `west-zephyr.yml` file of the Golioth SDK is _ignored_.
```

Comment:
```suggestion
These dependencies are something that you need to handle yourself since [_West_ ignores projects that have already been defined in other files](https://docs.zephyrproject.org/latest/develop/west/manifest.html#manifest-imports). E.g., if you define _Zephyr_ as a project in your manifest file, and _after_ add [Golioth's Zephyr SDK](https://docs.golioth.io/firmware/zephyr-device-sdk/), then the _Zephyr_ project in the `west-zephyr.yml` file of the Golioth SDK is _ignored_.
```